kgserver.js

```diff
@@ -760,6 +760,85 @@ app.get('/stylesheet', function (req, res) {
 
 });
 
+app.get('/layout-constraints', function (req, res)  {
+
+  res.setHeader('Access-Control-Allow-Origin', '*');
+
+  const constraintIRI = req.query.constraint ;
+
+  let store = $rdf.graph();
+  const constraint = $rdf.sym(utf8ToUnicode(constraintIRI));
+  const fetcher = createRdfFetcher(store);
+
+  fetcher.load(fetchableURI(constraintIRI)).then(response => {
+    let constraints = store.each(constraint, BROWSER("hasConstraint"), undefined);
+    let output = {
+      constraints: []
+    }
+    let classesToApplyConstraint = [];
+    let parentChildConstraint = {
+      childSelector: "",
+      edgeSelector: ""
+    }
+    Promise.all(
+      constraints.map(
+        function (constraint)  {
+          return new Promise((resolve, reject) => {
+            fetcher.load(fetchableURI(constraint.value)).then(response => {
+              let constraintType = unicodeToUTF8(constraint.value).match(/layout-constraints\/(.*)/)[1];
+              let constraintID = constraintType.match(/\/(.*)/)[1]
+              constraintType = constraintType.match(/(.*)\//)[1]
```

Comment:
IRIs shall be treated as unique identifiers only. You should not parse them as strings as sometimes you may not be able to choose their structure.

You should set the type and id explicitly as triplets with the IRI as a subject.